crates/control/src/time_to_reach_kick_position.rs

```diff
@@ -0,0 +1,136 @@
+use color_eyre::Result;
+use nalgebra::Point2;
+use spl_network_messages::Team;
+use std::{f32::consts::PI, time::Duration};
+
+use crate::behavior::walk_to_pose::WalkPathPlanner;
+use context_attribute::context;
+use framework::{AdditionalOutput, MainOutput};
+use types::{
+    configuration::{Behavior, Dribbling, PathPlanning},
+    FieldDimensions, GameControllerState, PathObstacle, PathSegment, WorldState,
+};
+#[context]
+pub struct CycleContext {
+    pub world_state: Input<WorldState, "world_state">,
+    pub time_to_reach_kick_position: PersistentState<Duration, "time_to_reach_kick_position">,
+    pub field_dimensions: Parameter<FieldDimensions, "field_dimensions">,
+    pub configuration: Parameter<Behavior, "behavior">,
+    pub parameters: Parameter<Behavior, "behavior">,
+    pub path_obstacles: AdditionalOutput<Vec<PathObstacle>, "time_to_reach_obstacles">,
+    pub stand_up_back_remaining_duration: Input<Duration, "stand_up_back_remaining_duration">,
+    pub stand_up_front_remaining_duration: Input<Duration, "stand_up_front_remaining_duration">,
+}
+
+#[context]
+pub struct CreationContext {}
+
+#[context]
+pub struct MainOutputs {
+    pub time_to_reach_kick_position: MainOutput<Option<Duration>>,
+}
+
+pub struct TimeToReachKickPosition {}
+
+impl TimeToReachKickPosition {
+    pub fn new(_: CreationContext) -> Result<Self> {
+        Ok(Self {})
+    }
+
+    pub fn cycle(&mut self, mut context: CycleContext) -> Result<MainOutputs> {
+        let path = self.plan_path(
+            context.world_state,
+            &context.configuration.path_planning,
+            &context.configuration.dribbling,
+            &mut context.path_obstacles,
+            context.field_dimensions,
+        );
+
+        let walk_time = path
+            .as_ref()
+            .map(|path| {
+                path.iter()
+                    .map(|segment: &PathSegment| {
+                        if matches!(segment, PathSegment::LineSegment(..)) {
```

Comment:
please use an `match` expression here. If other `PathSegment` types were to be added, this would map them to arc walking speed without any error.

Reply:
Done!